wgengine/userspace.go

```diff
@@ -73,6 +75,27 @@ type userspaceEngine struct {
 	// Lock ordering: wgLock, then mu.
 }
 
+// RouterGen is the signature for a function that creates a
+// router.Router.
+type RouterGen func(logf logger.Logf, wgdev *device.Device, tundev tun.Device) (router.Router, error)
+
+type EngineConfig struct {
+	// Logf is the logging function used by the engine.
+	Logf logger.Logf
+	// Tun is the TUN device used by the engine.
+	TUN tun.Device
```

Comment:
s/TUN/Tun/, TUN is not an acronym.

Reply:
I actually wonder about this: I've seen it (not necessarily correctly) treated as such, and was directed to use TUN [here](https://github.com/tailscale/tailscale/pull/358#discussion_r421548527). Should we use TUN or Tun everywhere? cc @bradfitz

Reply:
"tun" comes from tuntap, where it's shorthand for "tunnel", but not an initialism/acronym. So, by Go's style guide, I think it should be Tun everywhere, never TUN?

I think this is one where our code is definitely inconsistent right now. We can fix all of it up in a separate change too, we don't need to resolve the question right now.